### PR TITLE
feat(stream): add append_only in proto stream node

### PR DESCRIPTION
**Pull request by st1page**
## What's changed and what's your intention?
add append_only in proto stream node

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

**Review by TennyZhuang — APPROVED** [review]
LGTM

File: src/meta/src/stream/graph/stream_graph.rs [hunk 1]
```diff
@@ -581,6 +582,7 @@ where
                     fields: chain_node.upstream_fields.clone(),
                     operator_id: merge_node.operator_id,
                     identity: "MergeExecutor".to_string(),
+                    append_only: stream_node.append_only,
```
Comment (skyzh) [inline src/meta/src/stream/graph/stream_graph.rs:585]: Can we determine `append_only` of upstream node solely from the `StreamTableScan`'s `append_only` property?

**Review by skyzh — COMMENTED** [review]
Rest LGTM

File: src/meta/src/stream/graph/stream_graph.rs (hunk 1 above)
Comment (skyzh) [reply src/meta/src/stream/graph/stream_graph.rs:585]: I saw that `append_only` of `StreamTableScan` is always `false`... So I think we can follow the `append_only` property given by frontend for now. If there's issue after we implement the full append-only inference, let's fix it later.

**Review by skyzh — APPROVED** [review]

File: src/frontend/src/optimizer/plan_node/stream_table_scan.rs [hunk 2]
```diff
@@ -164,6 +165,7 @@ impl StreamTableScan {
             } else {
                 "".into()
             },
+            append_only: false, // TODO
```
Comment (skyzh) [inline src/frontend/src/optimizer/plan_node/stream_table_scan.rs:168]: should be `self.append_only`?

File: src/frontend/src/optimizer/plan_node/stream_table_scan.rs [hunk 3]
```diff
@@ -127,6 +127,7 @@ impl StreamTableScan {
                     pk_indices: pk_indices.clone(),
                     input: vec![],
                     fields: vec![], // TODO: fill this later
+                    append_only: true,
```
Comment (skyzh) [inline src/frontend/src/optimizer/plan_node/stream_table_scan.rs:130]: For L117 (MergeNode), also should be `self.append_only`?

File: src/meta/src/stream/graph/stream_graph.rs (hunk 1 above)
Comment (st1page) [reply src/meta/src/stream/graph/stream_graph.rs:585]: In fact I think the information should be in the catalog of the mv...

**Pull request merged**